app/src/main/java/org/mifos/mobile/ui/update_password/UpdatePasswordScreen.kt

```diff
@@ -0,0 +1,230 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MifosOutlinedTextField
+import org.mifos.mobile.core.ui.component.MifosProgressIndicatorWithText
+import org.mifos.mobile.core.ui.component.MifosTopBar
+import org.mifos.mobile.utils.RegistrationUiState
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun UpdatePasswordScreen(
+    viewModel: UpdatePasswordViewModel = hiltViewModel(),
+    getBackToPreviousScreen: () -> Unit
+) {
+    var newPassword by rememberSaveable(stateSaver = TextFieldValue.Saver) {
+        mutableStateOf(
+            TextFieldValue("")
+        )
+    }
+    var confirmPassword by rememberSaveable(stateSaver = TextFieldValue.Saver) {
+        mutableStateOf(
+            TextFieldValue("")
+        )
+    }
+    var viewNewPassword by rememberSaveable {
+        mutableStateOf(false)
+    }
+    var viewConfirmPassword by rememberSaveable {
+        mutableStateOf(false)
+    }
+
+    var newPasswordError by rememberSaveable { mutableStateOf(false) }
+    var confirmPasswordError by rememberSaveable { mutableStateOf(false) }
+
+    var newPasswordErrorContent by rememberSaveable {
+        mutableStateOf("")
+    }
+    var confirmPasswordErrorContent by rememberSaveable {
+        mutableStateOf("")
+    }
+
+    val uiState by viewModel.updatePasswordUiState.collectAsStateWithLifecycle()
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val context = LocalContext.current
+
+    when (uiState) {
+        is RegistrationUiState.Loading -> {
+            MifosProgressIndicatorWithText(text = stringResource(id = R.string.progress_message_loading))
+        }
+
+        is RegistrationUiState.Error -> {
+            LaunchedEffect(key1 = true) {
+                Toast.makeText(context, R.string.could_not_update_password_error, Toast.LENGTH_SHORT).show()
+            }
+        }
+
+        is RegistrationUiState.Initial -> Unit
+
+        is RegistrationUiState.Success -> {
+            LaunchedEffect(key1 = true) {
```

Comment:
use snackbar instead of toast and replace `LaunchedEffect` with coroutinescope

Reply:
bro, I used `coroutineScope` instead of `LaunchedEffect` as you said. But when I created another composable and tried to handle `uiState` inside, it's showing me `snackbar` constantly whenever UI-changes. As an example when I type something in the textfield. Which leads to Ui-change

I did some research, it seems like- it happened because we used `coroutinescope`. Even docs suggest to use LaunchedEffect. [Docs](https://developer.android.com/develop/ui/compose/side-effects)
I